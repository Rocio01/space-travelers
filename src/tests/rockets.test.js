import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Rockets from '../components/rockets';
import '@testing-library/jest-dom/extend-expect';
import store from '../redux/configureStore';
import RocketsMock from './mocks/RocketsMock';

describe('rocket component functionality', () => {
  test('matches Rockets component snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render Rockets component correctly', () => {
    const component = render(
      <RocketsMock
        id="TestId"
        rocket_name="rocket one"
        description="description one"
        reserved={false}
      />,
    );
    expect(component.container).toHaveTextContent('rocket one');
    expect(component.container).toHaveTextContent('description one');
  });

  test('If reserved = true innertext = Cancel Booking', () => {
    const component = render(
      <RocketsMock
        id="TestId"
        rocket_name="rocket two"
        description="description one"
        reserved
      />,
    );
    expect(component.container).toHaveTextContent('Cancel Booking');
  });

  test('If reserved = false innertext = reserved rocket', () => {
    const component = render(
      <RocketsMock
        id="TestId"
        rocket_name="rocket two"
        description="description one"
        reserved={false}
      />,
    );
    expect(component.container).toHaveTextContent('reserved rocket');
  });
});
