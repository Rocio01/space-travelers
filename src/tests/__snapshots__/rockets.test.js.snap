// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Rockets component snapshot 1`] = `
<div
  className="row m-4"
>
  <div
    className="col-3"
  >
    <img
      alt="flickr_images"
      className="rockets-images"
    />
  </div>
  <div
    className="col-9"
  >
    <h5 />
    <span
      className="btn"
    >
      
    </span>
    <p />
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      reserved rocket
    </button>
  </div>
</div>
`;
